src/Microsoft.AspNetCore.Mvc.Razor/Internal/CompilerCacheResult.cs

```diff
@@ -75,5 +78,11 @@ public CompilerCacheResult(IList<IChangeToken> expirationTokens)
         /// Gets a value that determines if the view was successfully found and compiled.
         /// </summary>
         public bool Success { get; }
+
+        /// <summary>
+        /// Gets a Func to create the view.
+        /// </summary>
+        /// <remarks>This property is not available when <see cref="Success"/> is <c>false</c>.</remarks>
+        public Func<object> PageCreate { get; }
```

Comment:
Yes, Func<IRazorPage> is set as a property of ViewLocationCacheItem which arte entries of a cache in RazorViewEngine.cs